compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -383,6 +383,39 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         return None;
                     } else {
                         span = item_name.span;
+
+                        // issue #81576, elision of generic argument when no methode can be found in any implementation
+                        let mut ty_str_reported = ty_str.clone();
+                        if let ty::Adt(_, ref generics) = actual.kind() {
+                            if generics.len() > 0 {
+                                let candidate_numbers: usize = self
+                                    .autoderef(span, actual)
+                                    .map(|(ty, _)| {
+                                        if let ty::Adt(ref adt_deref, _) = ty.kind() {
+                                            self.tcx
+                                                .inherent_impls(adt_deref.did)
+                                                .iter()
+                                                .filter_map(|def_id| {
+                                                    self.associated_item(
+                                                        *def_id,
+                                                        item_name,
+                                                        Namespace::ValueNS,
+                                                    )
+                                                })
+                                                .count()
+                                        } else {
+                                            0
+                                        }
+                                    })
+                                    .sum();
+                                if candidate_numbers == 0 && unsatisfied_predicates.is_empty() {
+                                    if let Some((path_string, _)) = ty_str.split_once('<') {
+                                        ty_str_reported = path_string.to_string();
+                                    }
```

Comment:
Ideally we'd just take the name ident of the type and use that. I think `item_name.to_string()` would  work here.

String operations are brittle and can break as things change in the codebase.

Reply:
`item_name` is the method we are looking to call. I did not find a way to print the type without the generic parameter otherwise. Do you know a way to do it ?

Reply:
What is the type of `item_name` from which you get `span` from?

https://github.com/rust-lang/rust/pull/84221/files/6efa14b3add08188c5322db8694a8cbbea7851e5#diff-e8458f9222609acec4270bb72dcf235e7aa575268adff217d113052c3751c53cL385

I think that might have a string representation, and if not a field that is only the name with type `Ident`, which has a string representation of _only_ the identifier.

Reply:
Yes it is an Ident, but this does not represent the type. This is the indent of the method

For instance it is used here: 
https://github.com/ABouttefeux/rust/blob/120691c590c4309fda31994931b9a561b4249c33/compiler/rustc_typeck/src/check/method/suggest.rs#L422-L431